050-Kafka-Node-Pools.md

```diff
@@ -0,0 +1,832 @@
+### Backwards compatibility and migration
+
+When the `KafkaNodePool` feature gate is enabled, users do not need to change anything right away.
+They can keep using their existing `Kafka` resources without any changes.
+The operator will continue to use the _virtual node pool_ as described in the previous section.
+
+When they decide to start using the actual `KafkaNodePool` resources, they can either:
+* Deploy a brand new Kafka cluster with one or more `KafkaNodePool` resources
+* Convert their existing Kafka cluster to use node pools
+
+#### Converting existing Apache Kafka clusters
+
+To convert an existing Kafka cluster to use node pools, the user has to do the following two steps:
+
+1. Create a new `KafkaNodePool` named `kafka` with the `.spec.replicas` and `.spec.storage` matching the configuration from their `Kafka` custom resource.
+2. Add an annotation `strimzi.io/node-pools: enabled` to the `Kafka` custom resource
+
+The following examples shows the important parts of the conversion.
+An initial `Kafka` custom resource like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    version: 3.3.1
+    replicas: 3
+    listeners:
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+    config:
+      offsets.topic.replication.factor: 3
+      transaction.state.log.replication.factor: 3
+      transaction.state.log.min.isr: 2
+      default.replication.factor: 3
+      min.insync.replicas: 2
+      inter.broker.protocol.version: "3.3"
+    storage:
+      type: jbod
+      volumes:
+        - id: 0
+          type: persistent-claim
+          size: 1000Gi
+          class: fast-storage
+  # ...
+```
+
+Is equivalent to the following combination of `Kafka` and `KafkaNodePool` resources:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+  annotations:
+    strimzi.io/node-pools: enabled
+spec:
+  kafka:
+    version: 3.3.1
+    listeners:
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+      - name: tls
+        port: 9093
+        type: internal
+        tls: true
+    config:
+      offsets.topic.replication.factor: 3
+      transaction.state.log.replication.factor: 3
+      transaction.state.log.min.isr: 2
+      default.replication.factor: 3
+      min.insync.replicas: 2
+      inter.broker.protocol.version: "3.3"
+  # ...
+---
+
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: kafka
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  storage:
+    type: jbod
+    volumes:
+    - id: 0
+      type: persistent-claim
+      size: 1000Gi
+      class: fast-storage
+  # ...
+```
+
+Because the `KafkaNodePool` is named `kafka`, all the resources created by this configuration will be exactly the same and have the same names as today.
```

Comment:
Is it possible changing the name from `kafka` to something else after the migration?

Reply:
Well, it depends on how you define _renaming_. You cannot rename Kubernetes resources. So I would say no, you cannot rename it. But of course, once you migrated to Node Pools, you can for example create a new node pool `akfak`, migrate all data to it, and delete the `kafka` node pool. You are not required to keep using the `kafka` node pool forever. But it is a bit more complicated then renaming it.